src/main/java/uk/gov/hmcts/reform/juddata/configuration/DataSourceConfig.java

```diff
@@ -15,12 +18,42 @@
 @EnableTransactionManagement
 public class DataSourceConfig {
 
-    @Autowired
-    DataSource dataSource;
+    @Value("${spring.datasource.url}")
+    String url;
+
+    @Value("${spring.datasource.username}")
+    String userName;
+
+    @Value("${spring.datasource.password}")
+    String password;
+
+    @Value("${spring.datasource.min-idle}")
+    int idleConnections;
+
+    @Value("${spring.datasource.max-timout}")
+    int maxTimeOut;
+
+    @Value("${spring.datasource.idle-timeout}")
+    int idleTimeOut;
+
+    @Bean
+    public DataSource dataSource() {
+        DataSourceBuilder dataSourceBuilder = DataSourceBuilder.create();
+        dataSourceBuilder.driverClassName("org.postgresql.Driver");
+        dataSourceBuilder.url(url);
+        dataSourceBuilder.username(userName);
+        dataSourceBuilder.password(password);
+        HikariDataSource dataSource = (HikariDataSource) dataSourceBuilder.build();
+        dataSource.setMinimumIdle(idleConnections);
+        dataSource.setIdleTimeout(idleTimeOut);
+        dataSource.setMaxLifetime(maxTimeOut);
+        return dataSource;
+    }
+
```

Comment:
can you also add a log to print out Hikari connection pool statistics.. we can see whats going wrong on case there is an issue?

Reply:
print the number of active, idle connections and the pool size